sootup.core/src/main/java/sootup/core/graph/DominanceFinder.java

```diff
@@ -39,19 +42,22 @@ public class DominanceFinder {
 
   public DominanceFinder(StmtGraph<?> blockGraph) {
 
-    // assign each block a integer id, startBlock's id must be 0
-    blocks = new ArrayList<>(blockGraph.getBlocks());
+    // we're locked into providing a List<BasicBlock<?>>, not a List<? extends BasicBlock<?>>, so
+    // we'll use the block iterator directly (which provides this type) rather than
+    // #getBlocksSorted.
+    blocks =
```

Comment:
as mentioned in the comment, this seems like an unfortunate api clash between what `blockGraph.getBlocksSorted()` returns (`List<? extends BasicBlock<?>>`) and what `DominanceFinder#getIdxToBlock` returns (`List<BasicBlock<?>>`). 

I didn't want to make a breaking api change in this pr (though if you're okay with it, that might be a good alternative!), so I've used `blockGraph.getBlockIterator()` here instead, which has the desired return type.